File: test/fixtures/node-superagent/get_referer.js
import request from 'superagent';

request
  .get('http://localhost:28139')
  .set('X-Requested-With', 'XMLHttpRequest')
  .set('User-Agent', 'SimCity')
  .set('Referer', 'https://website.com')
  .then(res => {
     console.log(res.body);
  });

File: test/fixtures/kotlin-ktor/get_referer.kt
import io.ktor.client.HttpClient
import io.ktor.client.request.get
import io.ktor.client.request.header
import io.ktor.client.statement.bodyAsText
import io.ktor.http.isSuccess
import kotlinx.coroutines.runBlocking

val client = HttpClient()

runBlocking {
    val response = client.get("http://localhost:28139") {
        header("X-Requested-With", "XMLHttpRequest")
        header("User-Agent", "SimCity")
        header("Referer", "https://website.com")
    }
    if (!response.status.isSuccess()) throw Exception("Request failed with status code ${response.status}")

    println(response.bodyAsText())
}
